tmt/hardware.py

```diff
@@ -507,7 +507,8 @@ def _from_specification(
             as_quantity: bool = True,
             as_cast: Optional[Callable[[str], ConstraintValueT]] = None,
             original_constraint: Optional['Constraint[Any]'] = None,
-            allowed_operators: Optional[list[Operator]] = None
+            allowed_operators: Optional[list[Operator]] = None,
+            unit: Optional[Any] = "bytes"
```

Comment:
How about calling it a `default_unit`? Seeing `unit`, one might suspect there will be some conversion performed, to enforce this unit, which is not what's going to happen.